common/k8s/src/middleware/metadata.rs

```diff
@@ -148,38 +166,40 @@ impl Middleware for K8sMetadata {
             if let Some(parse_result) = parse_container_path(file_name) {
                 let obj_ref =
                     ObjectRef::new(&parse_result.pod_name).within(&parse_result.pod_namespace);
-                if let Some(pod) = self.store.get(&obj_ref) {
-                    let meta_object =
-                        extract_image_name_and_tag(parse_result.container_name, pod.as_ref());
-                    if meta_object.is_some() && line.set_meta(json!(meta_object)).is_err() {
-                        trace!("Unable to set meta object{:?}", meta_object);
-                        return Status::Skip;
-                    }
-
-                    if let Some(ref annotations) = pod.metadata.annotations {
-                        if line
-                            .set_annotations(
-                                annotations
-                                    .iter()
-                                    .fold(KeyValueMap::new(), |acc, (k, v)| acc.add(k, v)),
-                            )
-                            .is_err()
-                        {
+                if let Some(ref store) = *self.store.lock().unwrap() {
```

Comment:
always success unwrap?

Reply:
Unwrap could panic if a previous holder of the lock panicked while holding the lock. There's not much else we can do in this situation, so panicking seems like an acceptable result.